hooks/validate_runnerpool.go

```diff
@@ -0,0 +1,97 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	constants "github.com/cybozu-go/github-actions-controller"
+	actionsv1alpha1 "github.com/cybozu-go/github-actions-controller/api/v1alpha1"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:failurePolicy=fail,matchPolicy=equivalent,groups=actions.cybozu.com,resources=runnerpools,verbs=create;update,versions=v1alpha1,name=runnerpool-hook.actions.cybozu.com,path=/runnerpool/validate,mutating=false,sideEffects=none,admissionReviewVersions={v1alpha1,v1,v1beta1}
+
+// RunnerPoolValidator is a validation webhook for RunnerPools.
+type RunnerPoolValidator struct {
+	client  client.Client
+	log     logr.Logger
+	decoder *admission.Decoder
+
+	repositoryNameList []string
+}
+
+// NewRunnerPoolValidator creates a validating webhook for RunnerPools.
+func NewRunnerPoolValidator(
+	k8sClient client.Client,
+	log logr.Logger,
+	decoder *admission.Decoder,
+	repositoryNameList []string,
+) http.Handler {
+	return &webhook.Admission{
+		Handler: RunnerPoolValidator{
+			client:             k8sClient,
+			log:                log,
+			decoder:            decoder,
+			repositoryNameList: repositoryNameList,
+		},
+	}
+}
+
+// Handle implements admission.Handler interface.
+func (v RunnerPoolValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Why don't use use `kubebuilder` to scaffold this?

Custom resources should implement webhooks with these interfaces:
- https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook#Validator
- https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook#Defaulter

and `kubebuilder create webhook` do this nicely.

Here is a good example:
https://github.com/cybozu-go/moco/blob/refactor-1/api/v1beta1/mysqlcluster_webhook.go